sound/soc/soc-topology.c

```diff
@@ -603,7 +605,14 @@ static int soc_tplg_kcontrol_bind_io(struct snd_soc_tplg_ctl_hdr *hdr,
 				sbe->get = ext_ops[i].get;
 		}
 
-		if (sbe->put && sbe->get)
+		if ((k->access & SNDRV_CTL_ELEM_ACCESS_TLV_READWRITE)
```

Comment:
this will be true for both READ_ONLY and WRITE_ONLY accesses as well. So it should be changed to 
(k->access & SNDRV_CTL_ELEM_ACCESS_TLV_READWRITE == SNDRV_CTL_ELEM_ACCESS_TLV_READWRITE)